pkg/reconciler/tektonwrapper/tektonwrapper.go

```diff
@@ -0,0 +1,271 @@
+			switch t.Status.State {
+			case v1alpha1.TektonWrapperStateUnprocessed:
+				unprocessedCount++
+			case v1alpha1.TektonWrapperStateInProgress:
+				activeCount++
+			case v1alpha1.TektonWrapperStateThrottled:
+				throttledCount++
+			case v1alpha1.TektonWrapperStateAbandoned:
+				fallthrough
+			case v1alpha1.DependencyBuildStateComplete:
+				doneCount++
+			default:
+				log.Info(fmt.Sprintf("GGM1 %s:%s with state %s of len %d", t.Namespace, t.Name, t.Status.State, len(strings.TrimSpace(t.Status.State))))
+				// create cannot update status
+				unprocessedCount++
+			}
+		}
+
+		log.Info(fmt.Sprintf("GGM2 raw %d active %d throttled %d abandoned/complete %d total %d",
+			unprocessedCount,
+			activeCount,
+			throttledCount,
+			doneCount,
+			totalCount))
+
+		if len(strings.TrimSpace(tw.Status.State)) == 0 {
+			// create cannot update status
+			tw.Status.State = v1alpha1.TektonWrapperStateUnprocessed
+		}
+
+		switch {
+		case (totalCount - doneCount) < hardPodCount:
+			// below hard pod count quota
+			break
+		case tw.Status.State == v1alpha1.TektonWrapperStateUnprocessed:
+			// big initial creation ... same effect as next case, but curious on the flow
+			r.eventRecorder.Eventf(&tw, corev1.EventTypeWarning, "ThrottledPipelineRun", "%s:%s", pr.Namespace, pr.Name)
+			tw.Status.State = v1alpha1.TektonWrapperStateThrottled
+			//stagger := 0
+			//if index < 60 {
+			//	stagger = int(math.Mod(60.0, float64(index)))
+			//} else {
+			//	stagger = int(math.Mod(float64(index), 60.0))
+			//}
+			//log.Info(fmt.Sprintf("GGM stagger %d", stagger))
+			//log.Info(fmt.Sprintf("GGM requeue %s", (tw.Spec.RequeueAfter + (time.Duration(stagger*10) * time.Millisecond)).String()))
+			return reconcile.Result{RequeueAfter: (tw.Spec.RequeueAfter /*+ (time.Duration(stagger*10) * time.Millisecond)*/)}, r.client.Status().Update(ctx, &tw)
+			// individual tw needs to be throttled since pas the limit
+		case ((hardPodCount - 2) <= activeCount) && tw.Status.State == v1alpha1.TektonWrapperStateThrottled:
+			// active too close, continue to requeue
+			timeout := tw.ObjectMeta.CreationTimestamp.Add(tw.Spec.AbandonAfter)
+			if timeout.After(time.Now()) {
+				return reconcile.Result{RequeueAfter: tw.Spec.RequeueAfter}, nil
+			}
+			tw.Status.State = v1alpha1.TektonWrapperStateAbandoned
+			r.eventRecorder.Eventf(&tw, corev1.EventTypeWarning, "AbandonedPipelineRun", "after throttling now past throttling timeout and have to abandon %s:%s", pr.Namespace, pr.Name)
+			return reconcile.Result{}, r.client.Status().Update(ctx, &tw)
+		}
+
+		break
+	}
+
+	// must clear out resource version for create
+	pr.ResourceVersion = ""
+	if pr.Labels == nil {
+		pr.Labels = map[string]string{}
+	}
+	pr.Labels[TektonWrapperId] = tw.Name
+	//TODO in theory we could set ourselves as an ownerref, but dependencybuild does that and currently makes assumptions
+	// that it is the only one; for now, we well let ab's/db's cleanup driver pr's cleanup.
```

Comment:
I don't think we should be owner, we should make this as transparent as possible imho.

Reply:
sounds good; I'll update the comment to reflect that we have decided not to set owner, be fully transparent, etc.